app/uk/gov/hmrc/softdrinksindustrylevystub/controllers/AuthorisedFilterAction.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.softdrinksindustrylevystub.controllers
+
+import javax.inject.Inject
+import play.api.http.HeaderNames
+import play.api.mvc._
+import play.api.mvc.Results.Unauthorized
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AuthorisedFilterAction @Inject()(val cc: ControllerComponents)
+  extends ActionBuilder[Request, AnyContent] with ActionFilter[Request] {
```

Comment:
do we need actionBuilder here?

Reply:
Let me double check and delete if not needed.

Reply:
This is needed as we compose two actions and at least one of them needs to extend action builder in order to be used in controllers